pyproject.toml

```diff
@@ -1,119 +1,106 @@
-    "pyhdf",  # hdf4 readers (MODIS)
-    "rasterio",  # GEOTIFF output
-    "psutil",  # Required for memory checks
-    "pyorbital",  # required by satpy
-    "ephem",  # Required for overpass predictor
-    "isodate",  # Required for overpass predictor
-    "satpy",  # efficiency improvements >= 0.33.1
-    "numexpr",  # for efficiency improvements
-    "scikit-image",  # Radius based center coverage checks
-    "jsonschema>=4.18.0a5",  # Provides validation for YAML plugins
-    "referencing",
-    "astropy", # Used for accounting leap seconds when converting epoch timestamps
-    "pyPublicDecompWt"  # Required to read SEVIRI data
-]
-
-[project.optional-dependencies]
-test = [
-    "pytest",  # Required for unit tests
-    "pytest-cov"  # Reports on test coverage
-]
-doc = [
-    "sphinx<7.2",  # Required for building documentation, v7.2 errors
-    "sphinx-design",  # Required for building documentation
-    "pydata-sphinx-theme",  # Required for building documentation
-]
-lint = [
-    "bandit",  # Syntax/security checking
-    "black",   # Syntax checking and reformatting
-    "pylint",  # Syntax checking
-    "flake8",  # Syntax checking
-    "flake8-rst-docstrings",  # numpy docstring checking
-    "flake8-docstrings",  # docstring checking
-    "flake8-rst",  # RST checking
-    "prettier",  # YAML, JSON, etc linting
-    "doc8",  # RST linting
-]
-debug = [
-    "ipython",
-]
-
-[project.entry-points."geoips.plugin_packages"]
-geoips = "geoips"
+keywords = ["GeoIPS", "Poetry"]
+packages = [{include = "geoips"}]  # not needed as tool.poetry.name is properly set already
+classifiers = [                # // list of PyPI trove classifiers to describe the project
+    "Programming Language :: Python :: 3", ]
 
-[project.entry-points.console_scripts]
-run_procflow = "geoips.commandline.run_procflow:main"
-convert_trackfile_to_yaml = "geoips.commandline.convert_trackfile_to_yaml:main"
-update_tc_tracks_database = "geoips.commandline.update_tc_tracks_database:main"
-xml_to_yaml_sector = "geoips.commandline.xml_to_yaml_sector:main"
-test_interfaces = "geoips.commandline.test_interfaces:main"
-list_available_modules = "geoips.commandline.list_available_modules:main"
-create_sector_image = "geoips.commandline.create_sector_image:main"
+# uses current version .postN and not incrementing
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+metadata = false    # don't include local-version hash; date doesn't appear by deafult
+#style = "pep440"    # pep440 is deafult; can rmeove later on; used for : keeping 0.0.8 as the default install until 0.0.9 is out
+style = "semver"
+#tag-branch = "feature-poetry"
+strict = true
+pattern = "(?x)^((?P<epoch>\\d+)!)?(?P<base>\\d+(\\.\\d+)*)([-._]?((?P<stage>[a-zA-Z]+)[-._]?(?P<revision>\\d+)?))?(\\+(?P<tagged_metadata>.+))?$"
+                         #           also: 3.6.0 incompatible with 0.20.3, but 0.21.0 works
+
+
+[tool.poetry-dynamic-versioning.substitution]
+files = ["*/_version.py"]
+
```

Comment:
@jsolbrig Do we know if this will work the same way as the current dynamic versioning (setuptools_scm)?

Reply:
It should. I had @SRIKANTH-KUMAR work pretty hard to make sure that it works the same way. Can you test it to be sure?

When you do `pip install -e .` it will update `_version.py`. 

If you just do `pip install .` it will not updated it because it's not being used from that location. It will be correct in the installed package, though. The same is true if you build a wheel for distribution.

Reply:
Okay, to further answer your question:

After playing with this some there were some problems that needed to be resolved. After resolving those problems, this works ALMOST identically to setuptools_scm with one exception. This method requires a template file to add versions to. This is going to be a problem because that file must be committed and can accidentally be updated.

See these issues:
- https://github.com/mtkennerly/poetry-dynamic-versioning/issues/147
- https://github.com/mtkennerly/poetry-dynamic-versioning/issues/146

For the time being it might be a good idea to add a CI check or a hook to ensure that `geoips/_version.py` is never updated.

Reply:
Hmm, is this even a problem, since it doesn't actually work with pip install -e?  I'm going to not worry about this right now, we will throw it in the "don't merge if modified" list so at least we won't accidntally merge to main.